### DISCUSSION TITLE
bidirectional many-to-many relation

### DISCUSSION THREAD

**Discussion by dominiquegarmier** [prisma/prisma#13214] [Q&A] [answered]
suppose i want to model friendships between users, i.e. if I am your friend you are also my friend. How would i model this in Prisma? I have only found things about unidirectional many-to-many in the [docs](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/self-relations)

My Idea was something along these lines:
```prisma
model User {
    id       String @id @default(uuid()) @db.Uuid
    friends  Friendship[] @relation("friendship")
}

model Friendship {
    left    User   @relation("friendship", fields: [leftId], references: [id])
    leftId  String @default(uuid()) @db.Uuid
    right   User   @relation("friendship", fields: [rightId], references: [id])
    rightId String @default(uuid()) @db.Uuid

    @@id([leftId, rightId])
}
```
But prisma complains about bot `right` and `left` pointing to the same friends. How can I solve this?

**Comment by dominiquegarmier** [comment]
for this to be possible I would have to be able to add a unique constraint that `leftId != rightId` and that the set `{leftId, rightId}` is unique not just `[leftId, rightId]`.

**Comment by nurul3101** [comment, marked as answer]
Hey @DominiqueGarmier 👋 
You could model your relations this way:

```prisma
model User {
  id             String       @id @default(uuid()) @db.Uuid
  friendsAsLeft  Friendship[] @relation("friendshipAsLeft")
  friendsAsRight Friendship[] @relation("friendshipAsRight")
}

model Friendship {
  left    User   @relation("friendshipAsLeft", fields: [leftId], references: [id])
  leftId  String @default(uuid()) @db.Uuid
  right   User   @relation("friendshipAsRight", fields: [rightId], references: [id])
  rightId String @default(uuid()) @db.Uuid

  @@id([leftId, rightId])
}
```
You are looking for [Disambiguating your relations](https://www.prisma.io/docs/concepts/components/introspection#disambiguating-relations)

**Reply by dominiquegarmier** [reply]
Hey!

So there is no way to have bidirectional (i.e. symmetrical) (aswell as many-to-many and self) relationships in prisma? In the example you mentioned I have to always keep two Friendship records to model a single symmetrical Friendship. Is there no way to tell Prisma that `leftId != rightId` (and other constraints...)? In that case the relationship would no longer be ambiguous, thus not needing both.

**Reply by nurul3101** [reply]
All the relations between models need to be on both sides of the model. It's a requirement from prisma's end.
As there are multiple relations between the same two models, you would need to add @relation attribute multiple times, I don't think there is any other way yet

**Reply by dominiquegarmier** [reply]
ok ty

**Reply by pperpich-cpi** [reply]
Hey @DominiqueGarmier, how did you end up querying a model like that? Let's say you wanted all the friends of user A, it seems like you would get a result set with lots of duplicates of user A. Right?

**Reply by dominiquegarmier** [reply]
> Hey @DominiqueGarmier, how did you end up querying a model like that? Let's say you wanted all the friends of user A, it seems like you would get a result set with lots of duplicates of user A. Right?

I never ended up implenting it like that. I believe my model ended up differentiating between followers and following (i.e. like on twitter).
